src/transformers/generation_beam_search.py

```diff
@@ -259,7 +259,7 @@ def process(
                         continue
                     if beam_indices is not None:
                         beam_index = beam_indices[batch_beam_idx]
-                        beam_index = beam_index + (next_index,)
+                        beam_index = beam_index + (batch_beam_idx,)
```

Comment:
`batch_beam_idx` -> beam idx considering all batches (between `num_beams*(batch_idx-1)` and `(num_beams*batch_idx)-1`)
`next_index` -> beam idx for the current batch (between 0 and `num_beams-1`)

Reply:
Ah yeah great point!